modules/gantt/app/services/gantt/default_query_generator_service.rb

```diff
@@ -82,37 +84,29 @@ def timeline_query(project)
         default_with_filter = add_columns(project)
 
         type_filter_values = milestone_ids(project).concat(phase_ids(project))
-        default_with_filter['f'] = if type_filter_values.any?
-                                     [{ 'n' => 'type', 'o' => '=', 'v' => type_filter_values }]
-                                   else
-                                     []
```

Comment:
The empty block was needed because there is otherwise a default filter to `status: open` (which we don't want for this query) causing the "timeline" query also to appear, although there are neither Milestones, nor a Phase.

<img width="500" alt="Bildschirmfoto 2024-01-31 um 14 36 57" src="https://github.com/opf/openproject/assets/7457313/bfabf11d-44a7-447d-8da2-1a7bc4cabe29">

Reply:
This is fixed as also the timelines menu is not shown if there are no type_filter_values